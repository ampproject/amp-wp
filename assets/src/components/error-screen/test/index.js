/**
 * External dependencies
 */
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ErrorScreen } from '..';

describe('ErrorScreen', () => {
	it('matches snapshot', () => {
		const wrapper = create(
			<ErrorScreen
				finishLinkLabel="Go to homepage"
				finishLinkUrl="http://my-exit-link.com"
				error={{
					message: 'The application failed',
					stack: 'ReferenceError: foo is not defined',
				}}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
});
