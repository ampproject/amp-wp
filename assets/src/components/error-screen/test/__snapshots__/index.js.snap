// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen matches snapshot 1`] = `
<div
  className="error-screen-container"
>
  <div
    className="error-screen components-panel"
  >
    <h1>
      Something went wrong.
    </h1>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "The application failed",
        }
      }
    />
    <p>
      Please submit details to our 
      <a
        href="https://wordpress.org/support/plugin/amp/"
        rel="noreferrer noopener"
        target="_blank"
      >
        support forum
      </a>
      .
    </p>
    <details>
      <summary>
        Details
      </summary>
      <pre>
        ReferenceError: foo is not defined
      </pre>
      <button
        aria-describedby={null}
        className="components-button components-clipboard-button is-secondary is-small"
        onCopy={[Function]}
        type="button"
      >
        Copy Error
      </button>
    </details>
    <p>
      <a
        href="http://my-exit-link.com"
      >
        Go to homepage
      </a>
    </p>
  </div>
</div>
`;
